theano/gpuarray/pool.py

```diff
@@ -15,6 +15,8 @@
     # To make sure theano is importable
     pass
 
+dtype_name_for_casting = 'int64'
```

Comment:
I don't feel like writing `dtype_name_for_casting` everywhere is an improvement.  It's longer and hides away the actual type used.  Since we use a fixed type in the C code we can also use a fixed type in the python code.

But this is a minor point and I will not block the PR for this.